### DISCUSSION TITLE
NDJson Support

### DISCUSSION THREAD

**Discussion by boris-42** [fastapi/fastapi#7292] [Questions] [answered]
In case when large json list of objects is send to backend any kind of bulk API. 
It make sense for multiple reasons to use NDJSON (http://ndjson.org/) instead of huge json list of objects. 

I didn't find anything in documentation/google about how to implement it... 

It would be nice to have standard way for it provided by framework...

**Comment by dmig** [comment]
You can implement this yourself: create either a streaming request or response handler and read/stream json objects record-by-record.

**Comment by NoahCardoza** [comment]
Is this something that you'd like to support within the module and would accept a PR for @tiangolo?

**Comment by ycd** [comment]
Does OpenAPI Specification allow to use of NDJSON? 

As far as I know even Swagger does not support it, you should return your response content as `text/plain`.

**Comment by tiangolo** [comment, marked as answer]
Thanks for the discussion here!

Yep, the problem is that OpenAPI doesn't support that, so there wouldn't be a straightforward way to support it here.

There are also several "competing" standards for JSON lines. I hope at some point to support more customizability of data types and parsing/serializing, and that would allow you to implement this for your use case, but for now, I'll pass on this. But thanks for the interest! ☕ 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
